From f6ba44d6f941e65f1d8981ab0df8c55f8dc1ea3f Mon Sep 17 00:00:00 2001
From: Benjamin Berg <benjamin.berg@intel.com>
Date: Fri, 14 Jun 2024 10:13:46 +0200
Subject: [PATCH] tests: Use more specific alloc_fail location

The test here is triggering the allocation failure in the static
wpa_config_parse_password() helper. Use this and decrease the count
instead of matching both wpa_config_set_quoted() and wpa_config_set()
and counting down based on that.

This is in preparation to fix the failure function matching to not do a
prefix match.

Signed-off-by: Benjamin Berg <benjamin.berg@intel.com>
---
 tests/hwsim/test_ap_hs20.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/hwsim/test_ap_hs20.py b/tests/hwsim/test_ap_hs20.py
index 9cf2d9040..35cb0b8fe 100644
--- a/tests/hwsim/test_ap_hs20.py
+++ b/tests/hwsim/test_ap_hs20.py
@@ -6018,7 +6018,7 @@ def test_ap_hs20_set_profile_failures(dev, apdev):
     interworking_select(dev[0], bssid, "home", freq=2412)
     dev[0].dump_monitor()
     dev[0].request("NOTE wpa_config_set(password)")
-    with alloc_fail(dev[0], 3, "wpa_config_set;interworking_set_eap_params"):
+    with alloc_fail(dev[0], 1, "wpa_config_parse_password;interworking_set_eap_params"):
         dev[0].request("INTERWORKING_CONNECT " + bssid)
         wait_fail_trigger(dev[0], "GET_ALLOC_FAIL")
     with alloc_fail(dev[0], 1, "interworking_set_hs20_params"):
-- 
2.43.0

